### PR TITLE
New resource/datasource `azurerm_data_share_dataset_blob_storage`

### PR DESCRIPTION
**Pull request by yupwei68**
Partially fix https://github.com/terraform-providers/terraform-provider-azurerm/issues/6480

Sub PR of : https://github.com/terraform-providers/terraform-provider-azurerm/pull/7064

resources:
`azurerm_data_share_dataset_blob_storage`
datashare dataset blob storage

data source:
`azurerm_data_share_dataset_blob_storage`
datashare dataset blob storage datasource

### CODE REVIEW

File: azurerm/internal/services/datashare/data_source_data_share_dataset_blob_storage.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+				Required:     true,
+				ValidateFunc: validate.DataShareID,
+			},
+
+			"container_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"storage_account_name": {
```
Comment (katbyte) [inline azurerm/internal/services/datashare/data_source_data_share_dataset_blob_storage.go:42]: would it make sense to have a storage account block? 🤔 
```
storage_account {
    name = "storage"
    resource_group_name = "rg"
}
```

File: azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_blob_storage_test.go [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+`, data.RandomInteger, data.Locations.Primary, data.RandomIntOfLength(12))
+}
+
+func testAccAzureRMDataShareDataSetBlobStorageFile_basic(data acceptance.TestData) string {
+	config := testAccAzureRMDataShareDataSetBlobStorage_template(data)
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_data_share_dataset_blob_storage" "test" {
+  name                                = "acctest-dsbds-file-%[2]d"
```
Comment (katbyte) [inline azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_blob_storage_test.go:203]: could we use capitals in every resource name possible?

File: azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_blob_storage_test.go [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+  name                = "acctest-dsa-%[1]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_data_share" "test" {
+  name       = "acctest_ds_%[1]d"
```
Comment (katbyte) [inline azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_blob_storage_test.go:166]: could we use capitals in every resource name possible>?

**Comment by yupwei68** [comment]
Hi @katbyte , corresponding changes have been pushed. Please continue reviewing.
resource:
```
=== RUN   TestAccAzureRMDataShareDataSetBlobStorageFile_basic
--- PASS: TestAccAzureRMDataShareDataSetBlobStorageFile_basic (342.79s)
=== RUN   TestAccAzureRMDataShareDataSetBlobStorageFolder_basic
--- PASS: TestAccAzureRMDataShareDataSetBlobStorageFolder_basic (407.71s)
=== RUN   TestAccAzureRMDataShareDataSetBlobStorageContainer_basic
--- PASS: TestAccAzureRMDataShareDataSetBlobStorageContainer_basic (395.06s)
=== RUN   TestAccAzureRMDataShareDataSetBlobStorage_requiresImport
--- PASS: TestAccAzureRMDataShareDataSetBlobStorage_requiresImport (406.29s)
```

data source:
```
=== RUN   TestAccDataSourceAzureRMDataShareDatasetBlobStorage_basic
=== PAUSE TestAccDataSourceAzureRMDataShareDatasetBlobStorage_basic
=== CONT  TestAccDataSourceAzureRMDataShareDatasetBlobStorage_basic
--- PASS: TestAccDataSourceAzureRMDataShareDatasetBlobStorage_basic (455.41s)
```

**Comment by jackofallops** [comment]
Tests for service pass:
image

**Pull request merged**